applications/GeoMechanicsApplication/custom_workflows/adaptive_time_incrementor.cpp

```diff
@@ -73,21 +75,31 @@ double AdaptiveTimeIncrementor::GetIncrement() const { return mDeltaTime; }
 
 void AdaptiveTimeIncrementor::PostTimeStepExecution(const TimeStepEndState& rResultantState)
 {
-    if (rResultantState.NonConverged() ||
-        (rResultantState.Converged() && (rResultantState.num_of_iterations == mMaxNumOfIterations))) {
-        mDeltaTime *= mReductionFactor;
-    } else if (rResultantState.Converged() && (rResultantState.num_of_iterations < mMinNumOfIterations)) {
-        mDeltaTime = std::min(mDeltaTime * mIncreaseFactor, mMaxDeltaTime);
+    if (rResultantState.Converged()) // it is converged also at the beginning of the cycles
+    {
+        if (rResultantState.num_of_iterations < mMinNumOfIterations) {
+            mDeltaTime = std::min(mDeltaTime * mIncreaseFactor, mMaxDeltaTime);
+        } else if (rResultantState.num_of_iterations == mMaxNumOfIterations) {
+            mDeltaTime *= mReductionFactor;
+        }
+        if (rResultantState.time < mEndTime) {
+            // Up-scaling to reach end_time without small increments
+            constexpr auto small_increment_factor = 1.0e-3;
+            if (auto small_time_increment = small_increment_factor * mDeltaTime;
```

Comment:
If I'm correct this could be const auto
```suggestion
            if (const auto small_time_increment = small_increment_factor * mDeltaTime;
```

Reply:
Yes, you are right. Anne has suggested to use the min value instead of it. So this const is gone now.